### ISSUE TITLE
Error while trying to `pub login`

### ISSUE THREAD

**Issue by tvolkert** [dart-lang/pub#2960]
```
$ pub --trace login
type 'String' is not a subtype of type 'Uri'
package:pub/src/command/login.dart 46:48                                              LoginCommand.retrieveUserInfo.<fn>
===== asynchronous gap ===========================
dart:async                                                                            Future.catchError
package:pub/src/utils.dart 113:52                                                     captureErrors.wrappedCallback
package:stack_trace                                                                   Chain.capture
package:pub/src/utils.dart 126:11                                                     captureErrors
package:pub/src/command.dart 164:13                                                   PubCommand.run
package:args/command_runner.dart 196:27                                               CommandRunner.runCommand
package:pub/src/command_runner.dart 150:26                                            PubCommandRunner.runCommand
package:pub/src/command_runner.dart 138:18                                            PubCommandRunner.run
/opt/s/w/ir/cache/builder/src/third_party/dart/third_party/pkg/pub/bin/pub.dart 9:48  main
---- Log transcript ----
FINE: Pub 2.13.0-150.0.dev
FINE: Loading OAuth2 credentials.
FINE: Loading OAuth2 credentials.
FINE: Saving OAuth2 credentials.
IO  : Writing 1398 characters to text file /Users/tvolkert/.pub-cache/credentials.json.
IO  : HTTP GET https://accounts.google.com/.well-known/openid-configuration
    | user-agent: Dart pub 2.13.0-150.0.dev
IO  : HTTP response 200 OK for GET https://accounts.google.com/.well-known/openid-configuration
    | took 0:00:00.180988
    | cache-control: public, max-age=3600
    | last-modified: Thu, 16 Jan 2020 21:53:16 GMT
    | date: Wed, 14 Apr 2021 03:33:29 GMT
    | access-control-allow-origin: *
    | content-encoding: gzip
    | vary: Accept-Encoding
    | age: 8
    | content-type: application/json
    | x-xss-protection: 0
    | server: sffe
    | alt-svc: h3-29=":443"; ma=2592000,h3-T051=":443"; ma=2592000,h3-Q050=":443"; ma=2592000,h3-Q046=":443"; ma=2592000,h3-Q043=":443"; ma=2592000,quic=":443"; ma=2592000; v="46,43"
    | accept-ranges: bytes
    | content-length: 464
    | x-content-type-options: nosniff
    | expires: Wed, 14 Apr 2021 04:33:29 GMT
FINE: Saving OAuth2 credentials.
IO  : Writing 1398 characters to text file /Users/tvolkert/.pub-cache/credentials.json.
ERR : type 'String' is not a subtype of type 'Uri'
FINE: Exception type: _TypeError
ERR : package:pub/src/command/login.dart 46:48                                              LoginCommand.retrieveUserInfo.<fn>
    | ===== asynchronous gap ===========================
    | dart:async                                                                            Future.catchError
    | package:pub/src/utils.dart 113:52                                                     captureErrors.wrappedCallback
    | package:stack_trace                                                                   Chain.capture
    | package:pub/src/utils.dart 126:11                                                     captureErrors
    | package:pub/src/command.dart 164:13                                                   PubCommand.run
    | package:args/command_runner.dart 196:27                                               CommandRunner.runCommand
    | package:pub/src/command_runner.dart 150:26                                            PubCommandRunner.runCommand
    | package:pub/src/command_runner.dart 138:18                                            PubCommandRunner.run
    | /opt/s/w/ir/cache/builder/src/third_party/dart/third_party/pkg/pub/bin/pub.dart 9:48  main
---- End log transcript ----
```

**Comment by tusharojha** [comment]
I think this is already fixed in e87b7b6.

**Comment by mit-mit** [comment]
Closing as assumed fixed; @tvolkert let us please know if you see this again
